src/main/java/com/palantir/docker/compose/configuration/DockerComposeFiles.java

```diff
@@ -38,10 +40,15 @@ public static DockerComposeFiles from(String... dockerComposeFilenames) {
                 .map(File::new)
                 .collect(toList());
         validateAtLeastOneComposeFileSpecified(dockerComposeFiles);
-        validateComposeFilesExist(dockerComposeFiles);
         return new DockerComposeFiles(dockerComposeFiles);
     }
 
+    public DockerComposeFiles withAdditionalFile(File composeFile) {
```

Comment:
Similar to the point I mentioned previously, I think all files should be known at some point, so there shouldn't be a need to have a `with...`, and similarly it gets into the whole "I'm a builder but can also represent partial state" if that makes sense.